gmso/tests/test_atom.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import unyt as u
+import pytest
+
+from gmso.core.atom import Atom
+from gmso.core.element import Lithium, Sulfur
+from gmso.core.atom_type import AtomType
+from gmso.tests.base_test import BaseTest
+from gmso.exceptions import GMSOError
+from pydantic import ValidationError
+
+
+class TestSite(BaseTest):
+    def test_new_site(self):
+        atom = Atom(name='atom')
+        assert atom.name == 'atom'
+        assert np.all(np.isnan(atom.position))
+        assert atom.position.units == u.nm
+
+    def test_dtype(self):
+        atom = Atom(name='atom', position=u.nm * np.zeros(3))
+        assert atom.position.dtype == float
+        assert isinstance(atom.position, u.unyt_array)
+        assert isinstance(atom.position, np.ndarray)
+
+    def test_name_none(self):
+        atom = Atom(name=None)
+        assert atom.name == 'Atom'
+
+    def test_setters_and_getters(self):
+        atom = Atom(name='Lithium', element=Lithium, charge=1, mass=6.941)
+
+        assert atom.name == 'Lithium'
+        assert atom.element == Lithium
+        assert atom.charge == 1*u.elementary_charge
+        assert atom.mass == 6.941*u.gram/u.mol
+
+        atom.name = 'Sulfur'
+        atom.element = Sulfur
+        atom.charge = -1
+        atom.mass=32.065
+
+        assert atom.name == 'Sulfur'
+        assert atom.element == Sulfur
+        assert atom.charge == -1*u.elementary_charge
+        assert atom.mass == 32.065*u.gram/u.mol
+
+        atom.mass = None
+        atom.charge = None
+
+    def test_set_with_atom_type(self):
+        lithium_type = AtomType(mass=6.941, charge=1)
+        atom = Atom(name='Lithium')
+        atom.atom_type = lithium_type
+
+        assert atom.charge == 1*u.elementary_charge
+        assert atom.mass == 6.941*u.gram/u.mol
+
+    @pytest.mark.parametrize('position', [[0.0, 0.0],
+        [0.0, 0.0, 0.0, 0.0, 0.0], [[0.0, 0.0], [0.0, 0.0]],
+        ['a', 'b', 'c'], ['a', 1, 1]])
```

Comment:
can we add one test where some of the values in the array are `NaN` and others are 0.0?

Reply:
We can do that but, I don't think we should do it in this PR, may be we should open this as an issue and add it there?